instrumentor/controllers/agentenabled/pods_webhook.go

```diff
@@ -202,6 +204,24 @@ func (p *PodsWebhook) injectOdigosInstrumentation(ctx context.Context, pod *core
 
 	for i := range pod.Spec.Containers {
 		container := &pod.Spec.Containers[i]
+
+		containerConfig := ic.Spec.GetContainerAgentConfig(container.Name)
+		if containerConfig == nil {
+			// no config is found for this container, so skip (don't inject anything to it)
+			continue
+		}
+		if !containerConfig.AgentEnabled || containerConfig.OtelDistroName == "" {
+			// container config exists, but no agent should be injected by webhook to this container
+			continue
+		}
+		if containerConfig.EnvInjectionMethod != nil &&
+			(*containerConfig.EnvInjectionMethod == common.LoaderFallbackToPodManifestInjectionMethod ||
+				*containerConfig.EnvInjectionMethod == common.LoaderEnvInjectionMethod) {
+			// lodaer has been migrated to use distros.
+			// we only process "pod-manifest" containers here.
+			continue
+		}
```

Comment:
this is ugly but temporary, we will get rid of it in future PR.
I don't want this one to change tons of things in one huge PR